python/tk_desktop/rpc.py

```diff
@@ -105,8 +205,9 @@ def run(self):
         Run the thread, accepting connections and then listening on them until
         they are closed.  Each message is a call into the function table.
         """
-        logger.debug("server listening on '%s'", self.pipe)
-        while True:
+        logger.debug("multi server listening on '%s'", self.pipe)
+        while self._is_closed is False:
```

Comment:
The tests exposed a situation where the server wouldn't shut down if the closed flag was raised while there wasn't a client available, so I've fixed this.